mailbox/plugin/quota-mailing/src/test/java/org/apache/james/mailbox/quota/mailing/search/QuotaSearcherTest.java

```diff
@@ -0,0 +1,170 @@
+
+@TestInstance(Lifecycle.PER_CLASS)
+public interface QuotaSearcherTest {
+
+    String DOMAIN = "bar.com";
+    String DOMAIN2 = "JoeBarTeam.com";
+    User USER = User.from("foo", Optional.of(DOMAIN));
+    User USER2 = User.from("foo2", Optional.of(DOMAIN));
+    User USER3 = User.from("foo3", Optional.of(DOMAIN));
+    User USER4 = User.from("foo", Optional.of(DOMAIN2));
+    User USER5 = User.from("foo", Optional.of(DOMAIN));
+    Quota<QuotaSize> _1_ON_3_SIZE_QUOTA = Quota.<QuotaSize>builder().used(QuotaSize.size(10)).computedLimit(QuotaSize.size(30)).build();
+    Quota<QuotaSize> _2_ON_3_SIZE_QUOTA = Quota.<QuotaSize>builder().used(QuotaSize.size(20)).computedLimit(QuotaSize.size(30)).build();
+    Quota<QuotaCount> _1_ON_4_COUNT_QUOTA = Quota.<QuotaCount>builder().used(QuotaCount.count(1)).computedLimit(QuotaCount.count(4)).build();
+    Quota<QuotaCount> _2_ON_4_COUNT_QUOTA = Quota.<QuotaCount>builder().used(QuotaCount.count(2)).computedLimit(QuotaCount.count(4)).build();
+    Quota<QuotaCount> _3_ON_4_COUNT_QUOTA = Quota.<QuotaCount>builder().used(QuotaCount.count(3)).computedLimit(QuotaCount.count(4)).build();
+    Instant INSTANT = Instant.ofEpochMilli(45554);
+    QuotaThresholdChangedEvent _1_ON_3_SIZE_EVENT = new QuotaThresholdChangedEvent(
+            EventId.fromSerialized(1),
+            HistoryEvolution.noChanges(),
+            HistoryEvolution.noChanges(),
+            _1_ON_3_SIZE_QUOTA,
+            _1_ON_4_COUNT_QUOTA,
+            UserQuotaThresholds.Id.from(USER));
+    QuotaThresholdChangedEvent _2_ON_3_SIZE_EVENT = new QuotaThresholdChangedEvent(
+            EventId.fromSerialized(2),
+            HistoryEvolution.noChanges(),
+            HistoryEvolution.noChanges(),
+            _2_ON_3_SIZE_QUOTA,
+            _1_ON_4_COUNT_QUOTA,
+            UserQuotaThresholds.Id.from(USER2));
+    QuotaThresholdChangedEvent _2_ON_4_COUNT_EVENT = new QuotaThresholdChangedEvent(
+            EventId.fromSerialized(3),
+            HistoryEvolution.noChanges(),
+            HistoryEvolution.noChanges(),
+            _1_ON_3_SIZE_QUOTA,
+            _2_ON_4_COUNT_QUOTA,
+            UserQuotaThresholds.Id.from(USER3));
+    QuotaThresholdChangedEvent _3_ON_4_COUNT_EVENT = new QuotaThresholdChangedEvent(
+            EventId.fromSerialized(4),
+            HistoryEvolution.noChanges(),
+            HistoryEvolution.noChanges(),
+            _1_ON_3_SIZE_QUOTA,
+            _3_ON_4_COUNT_QUOTA,
+            UserQuotaThresholds.Id.from(USER4));
+    QuotaThresholdChangedEvent _2_ON_3_SIZE_3_ON_4_COUNT_EVENT = new QuotaThresholdChangedEvent(
+            EventId.fromSerialized(5),
+            HistoryEvolution.noChanges(),
+            HistoryEvolution.noChanges(),
+            _2_ON_3_SIZE_QUOTA,
+            _3_ON_4_COUNT_QUOTA,
+            UserQuotaThresholds.Id.from(USER5));
+
+    @BeforeAll
+    default void setup(EventStore eventStore) {
+        eventStore.append(_1_ON_3_SIZE_EVENT);
```

Comment:
The tests should rely on messages being appended into a **MailboxManager** to let a real choice to the implementations.